frontend/src/utils/date.ts

```diff
@@ -45,7 +45,15 @@ export const normalizeDateToUtc = (date: Date | null) => {
   return new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()))
 }
 
-export const loadDateFromNormalizedDate = (date: Date | null) => {
+export const loadDateFromNormalizedDate = (date: string | Date | null) => {
+  if (typeof date === 'string') {
+    const parsedDate = parseISO(date)
+    return new Date(
+      parsedDate.getUTCFullYear(),
+      parsedDate.getUTCMonth(),
+      parsedDate.getUTCDate(),
+    )
+  }
   if (!date) return date
   return new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate())
```

Comment:
/nit Would this be terser?

```suggestion
  if (!date) return date
  const parsedDate = typeof date === 'string' ? parseISO(date) : date
  return new Date(
    Date.UTC(
      parsedDate.getFullYear(),
      parsedDate.getMonth(),
      parsedDate.getDate(),
    ),
  )```